contracts/commands/AaveStoplLossCommand.sol

```diff
@@ -0,0 +1,307 @@
+            (StopLossTriggerData)
+        );
+
+        // TODO legit caller write here and check in closePosition
+        console.log("msg sender", msg.sender);
+        console.log("command execute address this", address(this));
+        trustedCaller = stopLossTriggerData.positionAddress;
+        IAccountImplementation(stopLossTriggerData.positionAddress).execute(
+            address(this),
+            executionData
+        );
+        trustedCaller = address(0);
+    }
+
+    function isTriggerDataValid(
+        bool continuous,
+        bytes memory triggerData
+    ) external pure override returns (bool) {
+        StopLossTriggerData memory stopLossTriggerData = abi.decode(
+            triggerData,
+            (StopLossTriggerData)
+        );
+
+        return
+            !continuous &&
+            stopLossTriggerData.slLevel > 100 &&
+            (stopLossTriggerData.triggerType == 10 || stopLossTriggerData.triggerType == 11);
+    }
+
+    function closePosition(
+        EarnSwapData.SwapData calldata exchangeData,
+        AaveData memory aaveData,
+        AddressRegistry calldata addressRegistry
+    ) external {
+        DataTypes.ReserveData memory collReserveData = lendingPool.getReserveData(
+            aaveData.collateralTokenAddress
+        );
+        DataTypes.ReserveData memory debtReserveData = lendingPool.getReserveData(
+            aaveData.debtTokenAddress
+        );
+        uint256 totalToRepay = IERC20(debtReserveData.variableDebtTokenAddress).balanceOf(
+            aaveData.borrower
+        );
+        uint256 totalDebt = IERC20(collReserveData.aTokenAddress).balanceOf(aaveData.borrower);
+        IERC20(collReserveData.aTokenAddress).approve(addressRegistry.aaveStopLoss, totalDebt);
+        {
+            CloseData memory closeData;
+
+            address[] memory debtTokens = new address[](1);
+            debtTokens[0] = address(aaveData.debtTokenAddress);
+            uint256[] memory amounts = new uint256[](1);
+            amounts[0] = (101 * totalToRepay) / 100;
+            uint256[] memory modes = new uint256[](1);
+            modes[0] = uint256(0);
+
+            closeData.receiverAddress = addressRegistry.aaveStopLoss;
+            closeData.assets = debtTokens;
+            closeData.amounts = amounts;
+            closeData.modes = modes;
+            closeData.onBehalfOf = address(this);
+            closeData.params = abi.encode(
+                collReserveData.aTokenAddress,
+                aaveData.collateralTokenAddress,
+                addressRegistry.exchange,
+                aaveData.borrower,
+                exchangeData
+            );
+            closeData.referralCode = 0;
+            lendingPool.flashLoan(
+                closeData.receiverAddress,
+                closeData.assets,
+                closeData.amounts,
+                closeData.modes,
+                closeData.onBehalfOf,
+                closeData.params,
+                closeData.referralCode
```

Comment:
just 0 ?

Reply:
```
The referral program is currently in active and you can pass0 as thereferralCode.
In future for referral code to be active again, a governance proposal, with the list of unique referral codes for various integration must be passed via governance.
```

https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit